group_utils/README.md

```diff
@@ -0,0 +1,171 @@
+The group items are excluded.
+```javascript
+var groupAllStates = getAllMembersStates(group)
+```
+
+### getAllMembersLabels
+This function returns the __labels__ of direct and child group members as an array.
+The group items are excluded.
+```javascript
+var groupAllLabels = getAllMembersLabels(group)
+```
+
+### getAllMembersLabelsStrings
+This function returns the __labels__ of direct and child group members as a concatenated string.
+The group items are excluded.
+```javascript
+var groupAllLabelsString = getAllMembersLabelsString(group)
+```
+
+### __Arithmetic operations__
+Perform arithmetic operations on the states of Number members. 
+This functionality is the same as in the openHAB [group item definition](https://www.openhab.org/docs/configuration/items.html#derive-group-state-from-member-items).
+The following functions only use Items of type: Number, Dimmer or Rollershutter.
+Other Items and NULL or UNDEF states are ignored.
+
+### MembersSum
+Get the __SUM__ of direct members' number states.
+```javascript
+var sum = MembersSum(group)
+```
+
+### MembersAvg
+Get the __AVERAGE__ of direct members' number states.
+```javascript
+var avg = MembersAvg(group)
+```
+
+### MembersMin
+Get the __MINIMUM__ of direct members' number states.
+```javascript
+var min = MembersMin(group)
+```
+
+### MembersMax
+Get the __MAXIMUM__ of direct members' number states.
+```javascript
+var max = MembersMax(group)
+```
+
+### allMembersSum
+Get the __SUM__ of all (also child) members' number states.
+```javascript
+var sum = allMembersSum(group)
+```
+
+### allMembersAvg
+Get the __AVERAGE__ of all (also child) members' number states.
+```javascript
+var avg = allMembersAvg(group)
+```
+
+### allMembersMin
+Get the __MINIMUM__ of all (also child) members' number states.
+```javascript
+var min = allMembersMin(group)
+```
+
+### allMembersMax
+Get the __MAXIMUM__ of all (also child) members' number states.
+```javascript
+var max = allMembersMax(group)
+```
+
+### __Count operations__
+
+### countMembers
+Count the direct members matching a given comparison function. 
+Comparison is done through a user-passed function.
+The function returns how many members match the given comparison function.
+```javascript
+var counter = countMembers(group, compareFunc)
+// with the following example scheme:
+var counter = countMembers(group, function (i) { return i.getState() /* comparison operators */ })
+
+// examples
+var lightsOFF = countMembers('lights', function (i) { return i.getState() == OFF }) // the number of lights off
+var larg24 = GroupUtils.count('temperatures', return i.getState() > 24 ) // the number of temperatures higher than 24
+var smalEq24 = GroupUtils.count('temperatures', function (i) { return i.getState() <= 24 }) // the number of temperatures lower or equal than/to 24
+```
+Argument | Purpose | Required
+-|-|-
+`group` | The name of the group. | yes
+`compareFunc` | The comparison function, pass and use `i` to work with the item.  | yes
```

Comment:
Reword to something like "The comparison function which must accept one argument, an Item Object, and returns a boolean." We want to state explicitly what requirements they have to define the functions they pass to us. It's implicitly implied in the examples but I like the repetition.